cns/restserver/util.go

```diff
@@ -112,9 +113,17 @@ func (service *HTTPRestService) saveNetworkContainerGoalState(req cns.CreateNetw
 	existingNCStatus, ok := service.state.ContainerStatus[req.NetworkContainerid]
 	var hostVersion string
 	var existingSecondaryIPConfigs map[string]cns.SecondaryIPConfig //uuid is key
+	var existingNCVersion string
 	if ok {
 		hostVersion = existingNCStatus.HostVersion
 		existingSecondaryIPConfigs = existingNCStatus.CreateNetworkContainerRequest.SecondaryIPConfigs
+		existingNCVersion = existingNCStatus.VMVersion
+	} else {
+		// Host version is the NC version from NMAgent, set it -2 to indicate no result from NMAgent yet.
+		// Existing NC status will be set as -1 to indicate no version exist yet.
+		// Intentionally set default host version smaller than existing NC version.
+		hostVersion = "-2"
+		existingNCVersion = "-1"
```

Comment:
Again, set this to 0, in case it is not set. We must confirm the version we are getting from CRD (must always be non zero)

Reply:
2 reasons can't set it to 0.
1.  NC version started from 0 in DNC code.
2. Host version should not be the same as existing NC version in default mode.